### ISSUE TITLE
[Bug] Music does not play via Youtube.

### ISSUE THREAD

**Issue by ElectricDaFearox** [ItzDerock/simplevoicechat-music#4]
Whenever I try to play music, it searches and plays YouTube by default. However, probably due to an API change, it now does not play anything. 

Instead returns with a `Failed to Play Song: Sign In to confirm you are not a bot` 
https://i.imgur.com/0wZkh3m.png

**Comment by ItzDerock** [comment]
I just released v0.2.0 of this mod which updated the YouTube source which has fixed a few other YouTube-related issues. Can you give it a try and report back? 

https://github.com/ItzDerock/simplevoicechat-music/releases/tag/0.2.0

I'm unable to reproduce this bug, but I did find an open issue at https://github.com/lavalink-devs/youtube-source/issues/14 related to this.

**Comment by ElectricDaFearox** [comment]
After updating, I tried again, and it seems to be not fixed. 

Same response as previously.

**Comment by ItzDerock** [comment]
There's unfortunately not much I can do from my side. According to the issue I linked in my previous comment, YouTube have a list of IP ranges and ASNs (ie OVH, GalaxyGate, etc) that they flag as suspicious, and it seems that your hosting provider may fall under one of those ranges.

Lavaplayer removed authentication support previously too, so I can't add an option to log into a YouTube account to get past the sign in screen. They're not very adamant with adding authentication support back since its complex and I think YouTube have blocked accounts that tried to do this in the past too.

**Comment by ElectricDaFearox** [comment]
> There's unfortunately not much I can do from my side. According to the issue I linked in my previous comment, YouTube have a list of IP ranges and ASNs (ie OVH, GalaxyGate, etc) that they flag as suspicious, and it seems that your hosting provider may fall under one of those ranges.
> 
> Lavaplayer removed authentication support previously too, so I can't add an option to log into a YouTube account to get past the sign in screen. They're not very adamant with adding authentication support back since its complex and I think YouTube have blocked accounts that tried to do this in the past too.

Ah, that's unfortunate. Maybe Pebble Hosting might be using one of those services.

**Comment by lolcat-y** [comment]
Hey I'm getting the same issue with my discord bot Anytime I try to play specifically music with YouTube for example like a results or search results it returns this ``` 
Failed to Play Song: Sign In to confirm you are not a bot``` This is how I have my plugins and everything In my Application config 
lavalink:
  plugins:
    - dependency: "dev.lavalink.youtube:youtube-plugin:1.4.0" #
      repository: "https://maven.lavalink.dev/releases"
      snapshot: false
  pluginsDir: "./plugins"
  defaultPluginRepository: "https://maven.lavalink.dev/releases"
  defaultPluginSnapshotRepository: "https://maven.lavalink.dev/snapshots"
  server:
    password: "autocreativeco"
    sources:
      youtube: false

**Comment by ItzDerock** [comment]
@lolcat-y If you're having issues with your Discord Bot, leave a comment in https://github.com/lavalink-devs/youtube-source/issues/14 instead of here. This repo is for a Minecraft plugin that uses lavaplayer.

**Comment by ItzDerock** [comment]
Quick update on this, looks like some progress is being made upstream in https://github.com/lavalink-devs/youtube-source/pull/33

#### tldr of the implications:
A server administrator (someone with console access) will have to do a one-time setup by running something along the lines of  `/music authenticate`. The plugin will display a code and a URL. The admin will have to go to that URL, and enter the code. This logs the plugin into that google account. If you've ever set up YouTube on a TV, this is exactly like that. 

I'd also strongly suggest using a burner google account for this, since I don't want to have to deal with people losing their main account due to this plugin.

**Comment by ElectricDaFearox** [comment]
> Quick update on this, looks like some progress is being made upstream in [lavalink-devs/youtube-source#33](https://github.com/lavalink-devs/youtube-source/pull/33)
> #### tldr of the implications:
> 
> A server administrator (someone with console access) will have to do a one-time setup by running something along the lines of `/music authenticate`. The plugin will display a code and a URL. The admin will have to go to that URL, and enter the code. This logs the plugin into that google account. If you've ever set up YouTube on a TV, this is exactly like that.
> 
> I'd also strongly suggest using a burner google account for this, since I don't want to have to deal with people losing their main account due to this plugin.

I think this would be fine. My only concern is the ads. But I guess that's up the person who's making the pr.

**Comment by Omersaeed02** [comment]
Hey, so how exactly do I play videos from YouTube again? There is no such command as /music authenticate in Minecraft after the mod is added (using mod v0.2.0).

**Comment by ItzDerock** [comment]
Hi @Omersaeed02, the authentication feature hasn't been implemented yet -- we were waiting for the upstream YouTube library to implement authentication. 

Just checked, and it seems they merged in the changes two days ago, so sometime in the near future I'll have a look at that and add the required changes on the mod side and will publish a hopefully working version again.
